.github/workflows/main.yml

```
@@ -0,0 +1,255 @@
+      # opt-in with 'ci:agent-integration'
+      agent_integration_tests_ci:
+        description: 'Enable Agent Integration tests'
+        default: false
+        required: false
+        type: boolean
+      # disabled by default, not required for merge
+      # opt-in with 'ci:jdk-compatibility' tag on PR
+      jdk_compatibility_ci:
+        description: 'Enable JDK compatibility tests'
+        default: false
+        required: false
+        type: boolean
+      # disabled by default, not required for merge
+      # opt-in with 'ci:windows' tag on PR
+      windows_ci:
+        description: 'Enable Windows build & tests'
+        default: false
+        required: false
+        type: boolean
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - labeled
+      - ready_for_review
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+env:
+  JAVA_VERSION: 17
+  JAVA_DIST: temurin
+
+# limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/maven-goal
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: ${{ env.JAVA_DIST }}
+          command: ./mvnw install -DskipTests=true -Dmaven.javadoc.skip=true
+      - uses: ./.github/workflows/stash
+        with:
+          name: build
+          path: ${{ github.workspace }}
+
+  license:
+    name: License
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/unstash
+        with:
+          name: build
+          path: ${{ github.workspace }}
+      - name: install artifacts
+        uses: ./.github/workflows/maven-goal
+        with:
+          # Need to install packages in m2 local repository
+          command: ./mvnw install -DskipTests=true -Dmaven.javadoc.skip=true
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: ${{ env.JAVA_DIST }}
+      - name: mvn license
+        run: ./mvnw org.codehaus.mojo:license-maven-plugin:aggregate-third-party-report -Dlicense.excludedGroups=^co\\.elastic\\.
+
+  unit-tests:
+    name: Unit Tests
+    # Inputs aren't defined on some events
+    # When undefined, we need to emulate the default value
+    if: inputs.test_ci == true || inputs.test_ci == null
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/unstash
+        with:
+          name: build
+          path: ${{ github.workspace }}
+      - uses: ./.github/workflows/maven-goal
+        with:
+          command: ./mvnw test
```

```suggestion
          command: ./mvnw test -DargLine="-Delastic.apm.overwrite.config.docs=false"
```

This required argument was just added with a #3027.